library/ssl_tls13_server.c

```diff
@@ -1736,10 +1741,261 @@ static int ssl_tls13_handshake_wrapup( mbedtls_ssl_context *ssl )
     MBEDTLS_SSL_DEBUG_MSG( 2, ( "handshake: done" ) );
 
     mbedtls_ssl_tls13_handshake_wrapup( ssl );
+
+#if defined(MBEDTLS_SSL_SESSION_TICKETS)
+    mbedtls_ssl_handshake_set_state( ssl,
+                                     MBEDTLS_SSL_NEW_SESSION_TICKET );
+#else /* MBEDTLS_SSL_SESSION_TICKETS */
     mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_HANDSHAKE_OVER );
+#endif /* !MBEDTLS_SSL_SESSION_TICKETS */
     return( 0 );
 }
 
+/*
+ * Handler for MBEDTLS_SSL_NEW_SESSION_TICKET
+ */
+#define SSL_NEW_SESSION_TICKET_SKIP  0
+#define SSL_NEW_SESSION_TICKET_WRITE 1
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int ssl_tls13_write_new_session_ticket_coordinate( mbedtls_ssl_context *ssl )
+{
+    /* Check whether the use of session tickets is enabled */
+    if( ssl->conf->f_ticket_write == NULL )
```

Comment:
Spec: "This extension (Pre-Shared Key Exchange Modes extension) also restricts the modes for use with PSK resumption. Servers SHOULD NOT send NewSessionTicket with tickets that are not compatible with the advertised modes;".

Thus here we should probably also check the pre-shared key exchange modes proposed by the client. #6078 is merged now thus the reading of the  pre-shared key exchange modes extension on server side is available. But we should probably use the `tls13_kex_modes` field of handshake params just to store the modes proposed by the client (and check it here) and not using it to store eventually the selected key exchange mode like we do in `ssl_tls13_check_ephemeral_key_exchange()`. In `ssl_tls13_check_ephemeral_key_exchange()` we should probably like in the prototype rather use a `key_exchange` field, set it and check it wherever we currently set/use `tls13_kex_modes` but when parsing the pre-shared key exchange modes extension and here.

Reply:
Add 
```c
if( !mbedtls_ssl_tls13_some_psk_enabled( ssl ) )
    {
        MBEDTLS_SSL_DEBUG_MSG( 2, ( "psk or psk_ephemeral is not enabled" ) );
        return( SSL_NEW_SESSION_TICKET_SKIP );
    }
```
As my understand , it should be enough. right?

Reply:
I got failure after this change.  And I think `ssl_tls13_check_*_key_exchange` should be re-considered.  
I am prepare a new PR for it. #6069 and #6079 will preceding the new PR.

Reply:
#6106 is created for the failure

Reply:
#6112 is for key exchange mode check